hudi-utilities/src/main/java/org/apache/hudi/utilities/schema/SchemaRegistryProvider.java

```diff
@@ -81,11 +66,22 @@ private static Schema getSchema(String registryUrl) throws IOException {
 
   @Override
   public Schema getSourceSchema() {
-    return schema;
+    String registryUrl = config.getString(Config.SRC_SCHEMA_REGISTRY_URL_PROP);
```

Comment:
is it called only once during a run? Will it be an issue if it is called more than once and slightly different schema is returned?

Reply:
It is already handled in the code, in case different schema is returned, writeClient is recreated and new avro schemas are registered with spark configuration.

Reply:
this is called in every run and if we detect schema change, we register and recreate write client.